src/pybind/mgr/cephadm/serve.py

```diff
@@ -1693,6 +1680,46 @@ async def _get_container_image_info(self, image_name: str) -> ContainerInspectIn
         self.log.debug(f'image {image_name} -> {r}')
         return r
 
+    def _set_cephadm_command_timeout(self, timeout: Any) -> int:
+        def _safe_convert_to_int(t: Any) -> int:
+            if not isinstance(t, int):
+                try:
+                    t = int(t)
+                except Exception:
+                    # just use a default if we can't convert to an int
+                    self.log.error(f'Got cephadm command timeout {t} of type '
+                                   f'{type(t)} that could not be converted to '
+                                   'an integer. There was potentially an issue '
+                                   'with the value of the '
+                                   'mgr/cephadm/default_cephadm_command_timeout '
+                                   'setting. Changing to 900 seconds')
+                    t = 900
+                    self.mgr.default_cephadm_command_timeout = 900
+            return t
+
+        if not timeout:
+            # default global timeout if no timeout was passed
+            timeout = _safe_convert_to_int(self.mgr.default_cephadm_command_timeout)
+        else:
+            timeout = _safe_convert_to_int(timeout)
```

Comment:
You can probably drop the whole nested function by doing something like:
```
if not timeout:
    timeout = self.mgr.default_cephadm_command_timeout
# ensure timeout is an int
try:
    timeout_val = int(timeout)
except (TypeError, ValueError):
    # LOGGING GOES HERE
    timeout_val = 900 
    self.mgr.default_cephadm_command_timeout = timeout_val

# ... remaining stuff goes here ...
```

Reply:
good point. Something did feel off about the way it was written